### PR TITLE
Do not merge projections with nested TRY expression

### LINKED ISSUE #6750: Incorrect results due to improper inlining of arguments to TRY
The following query should fail:

```sql
SELECT TRY(x) IS NULL FROM (SELECT 1/y as x FROM (VALUES 1, 2, 3, 0, 4) t(y))
```

However, it succeeds and produces this incorrect output:
```
 _col0
-------
 false
 false
 false
 true
 false
(5 rows)
```

This is due to this check in MergeProjections, which only tests whether TRY appears at the root of an expression:

```java
private static boolean containsTry(ProjectNode node)
{
    return node.getAssignments().values().stream().anyMatch(TryExpression.class::isInstance);
}
```

### PR DESCRIPTION
**Pull request by kokosing**
Do not merge projections with nested TRY expression

Fixes #6750

### CODE REVIEW

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -5147,6 +5147,11 @@ public void testTry()
 
         // missing function argument
         assertQueryFails("SELECT TRY()", "line 1:8: The 'try' function must have exactly one argument");
+
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: Nit: remove extra blank line

**Comment by kokosing** [comment]
Comment addressed

**Comment by kokosing** [comment]
Travis is failing with unrelated issue in product tests.

**Comment by kokosing** [comment]
Rebased

**Comment by kokosing** [comment]
Rebased and comments addressed.

I posted two versions of finding `Try` expression. 
 - `AstNodeSearcher`
 - and `Stream` like as you suggested

Generally I prefer the first one as it uses call stack for expression traversing and it is easier to modify for complex cases.

**Comment by martint** [comment]
Merged with a couple of minor changes.

**Pull request closed without merging**